Framework/API/inc/MantidAPI/Algorithm.h

```diff
@@ -189,6 +193,163 @@ class MANTID_API_DLL Algorithm : public IAlgorithm,
   /// Override if the algorithm is not part of the Mantid distribution.
   const std::string helpURL() const override { return ""; }
 
+  template <typename T> void checkWorkspaceType() const {
```

Comment:
- Why public?
- Maybe a better name?
- Why member?
- Do we actually need a function for this? Is the compiler error from `std::is_convertible` hard to read?